docs/Operators.md

```diff
@@ -27310,6 +27311,69 @@ expect(
 </details>
 
 
+### <a name="RotaryEmbedding"></a><a name="rotaryembedding">**RotaryEmbedding**</a>
+
+  RotaryEmbedding is the implementation of rotary positional embeddings (RoPE) based on the paper https://arxiv.org/pdf/2104.09864.
+  The positions are represented as rotation matrices that are multiplied to query and key
+  before the inner product of query and key is taken.
+
+  Rotary embeddings are defined using the below functions:
+
+      def rotate_half(x):
+          """Rotates half the hidden dims of the input."""
+          x1 = x[..., : x.shape[-1] // 2]
+          x2 = x[..., x.shape[-1] // 2 :]
```

Comment:
I guess the op requires this dimension to be even? If so, worth clarifying in the spec.